lib/matplotlib/backend_bases.py

```diff
@@ -2559,6 +2674,120 @@ class NonGuiException(Exception):
     pass
 
 
+class WindowEvent(object):
+    def __init__(self, name, window):
+        self.name = name
+        self.window = window
+
+
+class WindowBase(cbook.EventEmitter):
+    """The base class to show a window on screen.
+
+    Parameters
+    ----------
+    title : str
+        The title of the window.
+    """
+
+    def __init__(self, title, **kwargs):
+        super(WindowBase, self).__init__(**kwargs)
```

Comment:
Okay, will do, I can't remember if I did this (combined with the ``get_window_title`` returning ``image`` on the base class) to satisfy a test, or whether just to avoid possible breakage with user code when they switch from using the old classes to using the new ones.

Anyway, if a problem occur, then I think we can find a better way to do this.